src/EditorFeatures/Core/Implementation/MissingImportTracking/MissingImportTrackingTaggerProvider.PasteCommandHandler.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis.AddImport;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Microsoft.VisualStudio.Utilities;
+using VSCommanding = Microsoft.VisualStudio.Commanding;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.MissingImportTracking
+{
+    internal partial class MissingImportTrackingTaggerProvider
+    {
+        [Export]
+        [Export(typeof(VSCommanding.ICommandHandler))]
+        [ContentType(ContentTypeNames.RoslynContentType)]
+        [Name("Missing Import Tracking Command Handler")]
+        [Order(After = PredefinedCommandHandlerNames.FormatDocument)]
+        [Order(Before = PredefinedCommandHandlerNames.Completion)]
+        internal partial class PasteCommandHandler : ForegroundThreadAffinitizedObject, IChainedCommandHandler<PasteCommandArgs>
+        {
+            public string DisplayName => "Missing Import Tracking"; // TODO: Add to resources
```

Comment:
oh. we didn't have paste commend handler before? who did formatting on pasted code before then?

Reply:
We have paste command handlers.  But the idea with command handlers is that instead of having a single handler that is responsible for all features, you can have multiple handlers for the same action, and they can do whatever is necessary *per feature*.

i.e. you don't want a single handler to handle Escape-Keys.  You want each feature to say "here is how i will handle escape keys".

Reply:
sure. confused by name. so we still have other command handler that handle paste command. it is AddMissingImportsCommandHander that also handle paste command.